compliance/collection/nodeinventorizer/caching_inventorizer_test.go

```diff
@@ -0,0 +1,436 @@
+			}
+			cs := *NewCachingScanner(mockScanner{}, path, 3*time.Second, 3*time.Second, maxBackoff, func(time.Duration) {})
+
+			actual := cs.readCacheState(path)
+
+			s.Equal(c.expectedBackoff, actual.BackoffDuration)
+		})
+	}
+}
+
+func (s *TestComplianceCachingSuite) TestReadCacheStateMaxBackoffOnFaultyWrap() {
+	path := fmt.Sprintf("%s/inventory-cache", s.T().TempDir())
+	maxBackoff := 42 * time.Second
+	brokenWrap := "{\"UnknownKey\":Value}"
+	err := os.WriteFile(path, []byte(brokenWrap), 0600)
+	s.NoError(err)
+	cs := *NewCachingScanner(mockScanner{}, path, 3*time.Second, 3*time.Second, maxBackoff, func(time.Duration) {})
+
+	actual := cs.readCacheState(path)
+
+	s.Equal(actual.BackoffDuration, maxBackoff)
+	s.Nil(actual.CachedInventory)
+}
+
+func (s *TestComplianceCachingSuite) TestReadInventoryWrapFaultyUnmarshal() {
+	inventoryCachePath := fmt.Sprintf("%s/inventory-cache", s.T().TempDir())
+	brokenWrap := "{\"CachedInventory\":42}"
+	err := os.WriteFile(inventoryCachePath, []byte(brokenWrap), 0600)
+	s.NoError(err)
+
+	actual, err := readInventoryWrap(inventoryCachePath)
+
+	s.Nil(actual)
+	s.Error(err)
+}
+
+func (s *TestComplianceCachingSuite) TestReadInventoryWrapDoesntExist() {
+	inventoryCachePath := fmt.Sprintf("%s/inventory-cache", s.T().TempDir())
+
+	actual, err := readInventoryWrap(inventoryCachePath)
+
+	s.Nil(actual)
+	s.NoError(err)
+}
+
+func (s *TestComplianceCachingSuite) TestScanWithoutExistingCacheWritesCache() {
+	initial := 8 * time.Second
+	cache := 2 * time.Minute
+	maxBackoff := 10 * time.Second
+	inventoryCachePath := fmt.Sprintf("%s/inventory-cache", s.T().TempDir())
+	cs := *NewCachingScanner(mockScanner{}, inventoryCachePath, cache, initial, maxBackoff, func(time.Duration) {})
+	nodeName := "testme"
+
+	// Check the directly returned result is correct
+	actual, err := cs.Scan(nodeName)
+	s.NoError(err)
+	s.Equal(nodeName, actual.GetNodeName())
+
+	// Check the cache that is written by cs.Scan is correct
+	w := s.readWrap(inventoryCachePath)
+	ci := s.wrapToInventory(w)
+
+	s.Equal(nodeName, ci.GetNodeName(), "The result should be written to cache correctly.")
+	s.Greater(w.CacheValidUntil.Unix(), time.Now().Unix(), "The cache should be valid to some time in the future.")
+}
+
+func (s *TestComplianceCachingSuite) TestScanWriteBackoffOnCacheFail() {
+	sleeper := mockSleeper{callCount: 0}
+
+	initial := 2 * time.Second // must be lower than BackoffDuration in file
+	cache := 5 * time.Second
+	maxBackoff := 10 * time.Second
+	inventoryCachePath := fmt.Sprintf("%s/inventory-cache", s.T().TempDir())
+	cs := *NewCachingScanner(mockScanner{}, inventoryCachePath, cache, initial, maxBackoff, sleeper.mockWaitCallback)
+	nodeName := "testme"
+
+	brokenWrap := "{\"UnknownKey\":Value}"
+	err := os.WriteFile(inventoryCachePath, []byte(brokenWrap), 0600)
+	s.NoError(err)
+
+	ci, err := cs.Scan(nodeName)
+
+	s.NoError(err)
+	s.Equal(10*time.Second, sleeper.receivedDuration, "Scan should have waited for maxBackoff")
```

Comment:
Using `maxBackoff` here would be ok because that's a variable introduced in this test (not in the code under test). Up to you.
```suggestion
	s.Equal(maxBackoff, sleeper.receivedDuration, "Scan should have waited for maxBackoff")
```